README.md

```diff
@@ -89,30 +89,48 @@ Next configure cmake. The preferred backend for MIOpen can be set using the `-DM
 cmake -DMIOPEN_BACKEND=OpenCL ..
 ```
 
-The above assumes that OpenCL is installed in one of the standard locations. If not, then manually set these two cmake variables: 
+The above assumes that OpenCL is installed in one of the standard locations. If not, then manually set these cmake variables: 
 
 ```
-cmake -DMIOPEN_BACKEND=OpenCL -DOPENCL_LIBRARIES=<opencl-library-path> -DOPENCL_INCLUDE_DIRS=<opencl-headers-path> ..
+cmake -DMIOPEN_BACKEND=OpenCL -DHIP_CXX_COMPILER=<hip-compiler-path> -DOPENCL_LIBRARIES=<opencl-library-path> -DOPENCL_INCLUDE_DIRS=<opencl-headers-path> ..
 ```
 
-And an example setting the dependency path:
+And an example setting the dependency path for an envirnment in ROCm 3.5 and later:
+```
+cmake -DMIOPEN_BACKEND=OpenCL -DHIP_CXX_COMPILER=/opt/rocm/llvm/bin/clang++ -DCMAKE_PREFIX_PATH=/some/local/dir ..
+```
+
+For ROCm 3.3 and earlier:
 ```
 cmake -DMIOPEN_BACKEND=OpenCL -DCMAKE_PREFIX_PATH=/some/local/dir ..
 ```
 
-### For HIP, run:
+### For the HIP backend on ROCm 3.3 and earlier, run:
 
 Set the C++ compiler to `hcc`.
 ```
 export CXX=<location-of-hcc-compiler>
-cmake -DMIOPEN_BACKEND=HIP -DCMAKE_PREFIX_PATH="<hip-installed-path>;<hcc-installed-path>;<clang-ocl-installed-path>" ..
+cmake -DMIOPEN_BACKEND=HIP -DCMAKE_PREFIX_PATH="<hip-installed-path>;<hcc-installed-path>;<clang-ocl-installed-path>;<miopen-dependency-path>" ..
+```
+
+An example cmake step can be:
 ```
+CXX=/opt/rocm/hcc/bin/hcc cmake -DMIOPEN_BACKEND=HIP -DCMAKE_PREFIX_PATH="/opt/rocm/hcc;/opt/rocm/hip;/some/local/dir" ..
+```
+
+### For the HIP backend on ROCm 3.5 and later, run:
+Set the C++ compiler to `clang++`.
+```
+export CXX=<location-of-clang++-compiler>
+cmake -DMIOPEN_BACKEND=HIP -DCMAKE_PREFIX_PATH="<hip-installed-path>;<rocm-installed-path>;<miopen-dependency-path>" ..
+```
+
 An example cmake step can be:
 ```
-CXX=/opt/rocm/hcc/bin/hcc cmake -DMIOPEN_BACKEND=HIP -DCMAKE_PREFIX_PATH="/opt/rocm/hcc;/opt/rocm/hip" ..
+CXX=/opt/rocm/llvm/bin/clang++ cmake -DMIOPEN_BACKEND=HIP -DCMAKE_PREFIX_PATH="/opt/rocm;/opt/rocm/hip;/some/local/dir" ..
```

Comment:
We already append the `CMAKE_PREFIX_PATH` with `/opt/rocm;/opt/rocm/hip`, so this is not necessary.